tests/cpp/core/LoggingUtilityTest.cpp

```diff
@@ -0,0 +1,199 @@
+
+}
+
+BOOST_AUTO_TEST_CASE( testWindowCount )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+
+    displayGroup->addContentWindow( window1 );
+    BOOST_CHECK( logger.get()->getWindowCount( ) == 1) ;
+    
+    displayGroup->removeContentWindow( window1 );
+    BOOST_CHECK( logger.get()->getWindowCount( ) == 1) ;
+
+}
+
+BOOST_AUTO_TEST_CASE( testRecentInteractionCounter )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    ContentWindowPtr window2 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+
+    BOOST_CHECK( logger.get()->getRecentInteractionCounter( ) == 0) ;
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+ 
+    displayGroup->addContentWindow( window1 );
+    displayGroup->focus( window1->getID() );
+    BOOST_CHECK( logger.get()->getRecentInteractionCounter( ) == 2) ;
+
+}
+
+BOOST_AUTO_TEST_CASE( testLastInteraction )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+    
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+    BOOST_CHECK( logger.get()->getLastInteraction( ) == "") ;
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+ 
+    displayGroup->addContentWindow( window1 );
+    displayGroup->focus( window1->getID() );
+    BOOST_CHECK( logger.get()->getLastInteraction( ) == "mode changed") ;
+
+}
+
+BOOST_AUTO_TEST_CASE( testJsonOutput )
+{
+    ContentPtr content( new DummyContent );
+    DisplayGroupPtr displayGroup( new DisplayGroup( wallSize ));
+    
+    ContentWindowPtr window1 = boost::make_shared<ContentWindow>( content );
+    ContentWindowPtr window2 = boost::make_shared<ContentWindow>( content );
+    std::unique_ptr<LoggingUtility> logger = make_unique<LoggingUtility>();
+
+    QObject::connect( displayGroup.get(), &DisplayGroup::contentWindowAdded,
+                      logger.get(), &LoggingUtility::contentWindowAdded );
+ 
+    std::unique_ptr<RestLogger> logContent;
+    logContent.reset( new RestLogger( *(logger.get()) ));
+    BOOST_CHECK( logContent->_toJSON() == defaultJson ) ;
```

Comment:
members with _ are supposed to be private. Please make it private and use the public logContent->toJSON()